### ISSUE TITLE
Web-Workspace -After renamed the workspace, you need to refresh the page to see the new name

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#12000]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to URL https://staging.new.expensify.com/
2. Login with new account
3. Click on the "Global action">New workspace
4. Rename new Workspace
5. Click "Save" button

## Expected Result:
Renamed Workspace displayed without refresh

## Actual Result:
After renamed the workspace, you need to refresh the page to see the new name

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.2.18.2

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**Email or phone of affected tester (no customers):** 

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/196772472-9c6035bd-2064-46d9-a847-9149a55f623c.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by ctkochan22** [comment]
This is reproducible on prod. Not a deploy blocker

**Comment by ctkochan22** [comment]
Looks like the workspace name is reverting after we get the message: 
<img width="1121" alt="image" src="https://user-images.githubusercontent.com/12980144/196824573-952f142c-cd61-4170-b93e-3fb3e8f23676.png">

**Comment by kbecciv** [comment]
We used Deploy Blocker label, because we can not reproduce it in production

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by ctkochan22** [comment]
This only happens if they change the name before you get an automatic message about guides

**Comment by ctkochan22** [comment]
Still happening. I think this is internal, need to test what keys are being updated when the message about the guidse comes though

**Comment by ctkochan22** [comment]
Testing locally, looking like it's tied to the guide message.

**Comment by ctkochan22** [comment]
Had to jump on higher priority issues, hoping to get to this week

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by neil-marcellini** [comment]
@ctkochan22 may I work on this one?

**Comment by ctkochan22** [comment]
Ugh yes. I want it but you should if you have the bandwidth. 

As I mentioned, I believe the report name is being updated with the oudated name when we send the follow up guide message: "Chat with your specialist..."

**Comment by neil-marcellini** [comment]
Thanks! I'll see if I can still reproduce the issue now that the changes in [this PR](https://github.com/Expensify/Web-Expensify/pull/35388) to push the updated workspace name are on production. I'll let you know if I have any questions.

**Comment by ctkochan22** [comment]
Cool I had some issues getting guides to work to successfully get that message sent in my local dev. But you should be able to just test if you can reproduce on prod/staging

**Comment by JmillsExpensify** [comment]
This should have a BZ member on it. I just re-applied the label to get assignment.

**Comment by neil-marcellini** [comment]
Yep, I'm still able to reproduce this issue. You don't need to use a brand new account, but it must have 0 workspaces. Next I will reproduce the issue on dev and figure out exactly why the concierge message is sending an outdated room name.

@ctkochan22 for reference here's an SO on [how to set up 121 assigned guides on dev](https://stackoverflow.com/c/expensify/questions/14339).

https://user-images.githubusercontent.com/26260477/200930024-72aedee4-4c51-44f6-a2f6-be590a990913.mov

**Comment by neil-marcellini** [comment]
I'm having trouble reproducing this issue on dev. I looked at the console logs on staging and I found that NewDot is receiving a Pusher event with the outdated report name after it receives the event with the updated report name.
I believe that this problematic push event is coming from Policy.create [here](https://github.com/Expensify/Web-Expensify/blob/cf6e14ae37542719cee69641b9fdd1b3b3c2954a/lib/PolicyAPI.php#L256) since it contains two updates, one is a merge on the policy and one is a merge collection on the report collection.

![image](https://user-images.githubusercontent.com/26260477/200948785-75ae689a-b965-46d7-9b1b-c933748455d1.png)

Here's proof that an outdated workspace name is sent in that second pusher event.
![image](https://user-images.githubusercontent.com/26260477/200949863-abb3198d-7a20-4c5f-81ef-805009f3ddfd.png)



Looking at [the logs](https://staging.expensify.com/_devportal/tools/logSearch/#sort=asc&size=10000&query=blob%3A%22Sending%20command%20response%20via%20Pusher%22%20AND%20blob%3A%22policy_2CF52754BC81F205%22%20AND%20email%3A%22neil.marcellini%2Bdw%40gmail.com%22%20AND%20timestamp%3A%5B2022-11-09T21%3A16%3A03%20TO%202022-11-10T23%3A59%5D) of pusher events sent for that policy I can see that the first one sent is for creating the policy, and the second one is for updating the name. So they are sent in the right order but they arrive in a different order, which is totally possible.

**Comment by neil-marcellini** [comment]
Yep that's definitely the issue, if you wait until the policy creation push event comes back before updating the workspace name then it works fine. I'm not really sure what to do to fix this, because we don't have control over which pusher event will reach the client first.

We could prevent the user from updating the workspace general settings until they have received the policy creation pusher response, but then that would break offline flows.

https://user-images.githubusercontent.com/26260477/200957488-b0cc20a4-3b8d-4c6f-8956-62a291af8627.mov

**Comment by neil-marcellini** [comment]
I [posted in Slack](https://expensify.slack.com/archives/C03TQ48KC/p1668036419585519) for feedback on possible solutions.

**Comment by neil-marcellini** [comment]
In that discussion we decided that this needs a design doc. I started the project process today, designing in private, and I made a lot of good progress. I started [this tracking issue](https://github.com/Expensify/Expensify/issues/242444) as well and this issue will need to hold on it.

**Comment by davidcardoza** [comment]
@neil-marcellini Should the conversation move on to the tracking issue if this issue is planned to go through the design doc process - or is there a reason to keep both open?

**Comment by JmillsExpensify** [comment]
I think we should keep both open. I added this issue to the tracker.

**Comment by neil-marcellini** [comment]
I'm going to hold this issue on the tracking issue and I'll give daily updates there.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)

**Comment by neil-marcellini** [comment]
I'm going to make this monthly while it's on HOLD.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Fourth week). Can we close this as its not repro?

**Comment by JmillsExpensify** [comment]
@neil-marcellini What do you think? Given that QA isn't able to reproduce this any long I'm inclined to close, or we need to update the reproduction steps so it's more clear how Applause can get the same result.

**Comment by neil-marcellini** [comment]
I tested on staging and it looks like it's no longer reproducible while online, which is kind of odd. I'm not sure how it was fixed  but that's good. However the issue is still reproducible when the workspace is created and renamed while offline. I will update the description.

**Comment by JmillsExpensify** [comment]
Thanks! I'll also take this over as part of the broader tracking initiative. Feel free to assign yourself to #12775 as well.

**Comment by JmillsExpensify** [comment]
Still holding and not a priority.

**Comment by JmillsExpensify** [comment]
Still far from being a priority.

**Comment by neil-marcellini** [comment]
We are still holding this on #12775. I'm suggesting that we start working on that issue again.

**Comment by JmillsExpensify** [comment]
Awesome that sounds great!

**Comment by neil-marcellini** [comment]
Holding, we have a potential solution for #12775.

**Comment by neil-marcellini** [comment]
Still holding, see the updates in the replay effect issue.
